beacon-chain/p2p/pubsub.go

```diff
@@ -0,0 +1,62 @@
+package p2p
+
+import (
+	"context"
+	"encoding/base64"
+
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	pubsub_pb "github.com/libp2p/go-libp2p-pubsub/pb"
+	"github.com/prysmaticlabs/prysm/shared/hashutil"
+)
+
+// JoinTopic will join PubSub topic, if not already joined.
+func (s *Service) JoinTopic(topic string, opts ...pubsub.TopicOpt) (*pubsub.Topic, error) {
+	if _, ok := s.joinedTopics[topic]; !ok {
+		topicHandle, err := s.pubsub.Join(topic, opts...)
+		if err != nil {
+			return nil, err
+		}
+		s.joinedTopics[topic] = topicHandle
+	}
+
+	return s.joinedTopics[topic], nil
+}
+
+// LeaveTopic closes topic and removes corresponding handler from list of joined topics.
+// This method will return error if there are outstanding event handlers or subscriptions.
+func (s *Service) LeaveTopic(topic string) error {
+	if t, ok := s.joinedTopics[topic]; ok {
+		if err := t.Close(); err != nil {
+			return err
+		}
+		delete(s.joinedTopics, topic)
+	}
+	return nil
+}
+
+// PublishToTopic joins (if necessary) and publishes a message to a PubSub topic.
+func (s *Service) PublishToTopic(ctx context.Context, topic string, data []byte, opts ...pubsub.PubOpt) error {
+	topicHandle, err := s.JoinTopic(topic)
+	if err != nil {
+		return err
+	}
+	return topicHandle.Publish(ctx, data, opts...)
+}
+
+// SubscribeToTopic joins (if necessary) and subscribes to PubSub topic.
+func (s *Service) SubscribeToTopic(topic string, opts ...pubsub.SubOpt) (*pubsub.Subscription, error) {
+	topicHandle, err := s.JoinTopic(topic)
+	if err != nil {
+		return nil, err
+	}
+	return topicHandle.Subscribe(opts...)
+}
+
+// Content addressable ID function.
+//
+// Loosely defined as Base64(sha2(data)) until a formal specification is determined.
+// Pending: https://github.com/ethereum/eth2.0-specs/issues/1528
```

Comment:
```suggestion
// ETH2 spec defines the message ID as:
//    message-id: base64(SHA256(message.data))
```

The pending issue has been resolved.